### ISSUE TITLE
Use your own pre-trained model for downstream segmentation

### ISSUE THREAD

**Issue by yyy1998-i** [facebookresearch/mae#204]
I would like to ask you a question, I pre-train on my medical image data set (6k) according to the official pre-training code, and then the downstream task is segmentation, but the fine-tuning results with my own pre-training model is always inferior to the author's pre-training model with imagenet, which is why, is my pre-training data is still too small?

**Comment by Jivitesh2001** [comment]
Hey,
Your dataset seems too small compared to Imagenet-1K. A suggestion - you should initialize your model using the pre-trained model weights of Imagenet and then continue pre-training on your medical image dataset, which you can use for your downstream task. Please let me know how this goes for you.

**Comment by yyy1998-i** [comment]
Ok, thank you very much for your reply. Next, I will try according to what you said. Thanks again.

> 嘿，与 Imagenet-1K 相比，您的数据集似乎太小了。建议 - 您应该使用 Imagenet 的预训练模型权重初始化模型，然后继续对医学图像数据集进行预训练，该数据集可用于下游任务。请让我知道这对你来说怎么样。

**Comment by yyy1998-i** [comment]
> Hey, Your dataset seems too small compared to Imagenet-1K. A suggestion - you should initialize your model using the pre-trained model weights of Imagenet and then continue pre-training on your medical image dataset, which you can use for your downstream task. Please let me know how this goes for you.

Ok, thank you very much for your reply. Next, I will try according to what you said. Thanks again

**Comment by Jivitesh2001** [comment]
Hello,
Have you tried this approach? and Can you tell me what were your results?

**Comment by yyy1998-i** [comment]
> Hello, Have you tried this approach? and Can you tell me what were your results?

Sorry for the late reply. This is my latest training situation, and it is still not as good as the weight of imagenet.
![7c58737620d28084707bb3ea6c8c735f](https://github.com/user-attachments/assets/b58befc0-d5df-4c4f-b6d0-368492ab629e)

**Comment by Jivitesh2001** [comment]
Hey!
Have you tried different augmentation techniques to oversample your medical dataset? I think the idea should be a) Load the ImageNet Pre-Trained Checkpoint, b) Then use the pre-trained checkpoint to fine-tune the model (which seems to be a classification model) on your dataset for your downstream task. Have you tried this approach?

**Comment by yyy1998-i** [comment]
> Hey! Have you tried different augmentation techniques to oversample your medical dataset? I think the idea should be a) Load the ImageNet Pre-Trained Checkpoint, b) Then use the pre-trained checkpoint to fine-tune the model (which seems to be a classification model) on your dataset for your downstream task. Have you tried this approach?

Next, I will try to use data enhancement (the previous training was in accordance with the official default Settings). In the pre-training process, I found that without using normalize,MAE's reconstruction effect is better, and the loss is also very low (0.0822). When fine-tuning, I used mean and std which I used to fine-tune the data set. I found that this also affected the effectiveness of the downstream classification task.

**Comment by Yummy198913** [comment]
Same question,  I tried to pre-train my own 2.5 million data based on MAE-G with 4 A100 and used the pre-trained weights to fine-tune downstream classification tasks. However, I found that the metrics remained unchanged every epoch. There is no operation in the code related to freezing parameters of classification network. Could this be caused by the model being too large and the pre-training data being too small?
What`s more, the reconstruction visualization during pre-training does not seem to be very effective. It appears that 2.5 million data are not sufficient for MAE-G to converge well. Additionally, the quality of this data is not as high as that of ImageNet-1k.

**Comment by liaoyt26** [comment]
> I would like to ask you a question, I pre-train on my medical image data set (6k) according to the official pre-training code, and then the downstream task is segmentation, but the fine-tuning results with my own pre-training model is always inferior to the author's pre-training model with imagenet, which is why, is my pre-training data is still too small?

Hi,
May I ask which segmentation decoder you used in the downstream segmentation fine-tuning part? I’m also working on fine-tuning for medical image segmentation. I tried using a UNet-like decoder with upsampling, but the performance was quite poor. I’d really appreciate your insights or suggestions.
